source/tutorials/installing-nixos-on-a-raspberry-pi.rst

```diff
@@ -0,0 +1,175 @@
+Installing NixOS on a Raspberry Pi
+==================================
+
+This tutorial assumes `Raspberry P 4 Model B with 4GB RAM <https://www.raspberrypi.org/products/raspberry-pi-4-model-b/>`_.
+
+Before starting this tutorial, make sure you have 
+`all necessary hardware <https://projects.raspberrypi.org/en/projects/raspberry-pi-setting-up/1>`_:
+
+- HDMI cable/adapter.
+- 8GB+ SD card.
+- SD card reader in case your machine doesn't have an SD slot.
+- Power cable for your RPi.
+- USB keyboard.
+
+.. note:: 
+
+  RPi 2 won't work since it doesn't support 64bit OS, while RPi 3 should work with more fiddling.
+
+
+Booting NixOS live image
+------------------------
+
+.. note:: Booting from a USB for vast majority of boards won't work without updating firmware. This tutorial boots from an SD card to avoid such hiccups.
+
+Prepare the Aarch64 image on your laptop:
+
+.. code:: shell-session
+
+  $ nix-shell -p wget zstd
+  $ wget https://hydra.nixos.org/build/142828023/download/1/nixos-sd-image-21.05pre288297.8eed0e20953-aarch64-linux.img.zst
+  $ unzstd -d nixos-sd-image-21.05pre288297.8eed0e20953-aarch64-linux.img.zst
+  $ dmesg -w
```

Comment:
I'd like to pin it so that I'm 100% that it works for the user. I'll set up tasks to update these for every release.

Maybe at the end of tutorial we should do `nixos-rebuild switch --upgrade`